xscen/extract.py

```diff
@@ -261,6 +261,29 @@ def extract_dataset(
             ),
             reverse=True,
         ):
+            if "time" in ds_ts and ensure_correct_time:
+                # Expected freq (xrfreq is the wanted freq)
+                expfreq = catalog[key].df.xrfreq.iloc[0]
+                # Check if we got the expected freq (skip for too short timeseries)
+                inffreq = xr.infer_freq(ds_ts.time) if ds_ts.time.size > 2 else None
+                if inffreq == expfreq:
+                    # Even the freq is correct, we ensure the correct "anchor" for daily and finer
+                    if expfreq in "DHTMUL":
+                        ds_ts["time"] = ds_ts.time.dt.floor(expfreq)
```

Comment:
I don't think this fixes the issue of variables that are not quite at the same hour?

Reply:
It doesn't fix the issue where different variables of the same dataset (after `to_dataset_dict`) have different hours, indeed. But the previous code didn't either. That case will trigger the first `ValueError` below.

Reply:
Alright. I can approve this PR as is and we'll look into that at a later date?

Reply:
Yep! It wasn't my plan here to fix this, only to accelerate it.